src/Component/Metadata/Resource/Factory/AttributesResourceMetadataCollectionFactory.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Component\Resource\Metadata\Resource\Factory;
+
+use Sylius\Component\Resource\Metadata\Operation;
+use Sylius\Component\Resource\Metadata\Operations;
+use Sylius\Component\Resource\Metadata\Resource as ResourceMetadata;
+use Sylius\Component\Resource\Metadata\Resource\ResourceMetadataCollection;
+use Sylius\Component\Resource\Reflection\ClassReflection;
+use Webmozart\Assert\Assert;
+
+final class AttributesResourceMetadataCollectionFactory implements ResourceMetadataCollectionFactoryInterface
+{
+    public function create(string $resourceClass): ResourceMetadataCollection
+    {
+        $resourceMetadataCollection = new ResourceMetadataCollection();
+
+        $attributes = ClassReflection::getClassAttributes($resourceClass);
+
+        foreach ($this->buildResourceOperations($attributes, $resourceClass) as $resource) {
+            $resourceMetadataCollection[] = $resource;
```

Comment:
Can we adjust this logic to following rules:
1. One class may have several resource attributes
2. Each resource attribute requires alias and this alias is unique among all resources
3. There is no possibility to define any operations on your own
4. To declare any operation, you have to add `Routing` attribute, which will require alias field to be declared and matched with any resource alias (the question is, should we allow to define routing for aliases from other classes)
To improve DX in the next iteration we can consider autogeneration of alias with the pattern: `app.your_resource`

Reply:
> Can we adjust this logic to following rules:
> 
>     1. One class may have several resource attributes

agree

> 
>     2. Each resource attribute requires alias and this alias is unique among all resources

for the required alias, I agree. 
For the unique alias, I'm not sure, I think we can use the same alias on an extended class to customize already defined operations in sub class.

> 
>     3. There is no possibility to define any operations on your own

Why? I think it's very interesting to allow them.

> 
>     4. To declare any operation, you have to add `Routing` attribute, which will require alias field to be declared and matched with any resource alias (the question is, should we allow to define routing for aliases from other classes)

For me, you always need a resource attribute to define operations. But the one attribute per line is interesting. Moreover, one of the main goal of this implementation is to be closer to API Platform. Thus, Sylius will be easier for beginners to learn. For now, a lot of Symfony users are struggling with the resource system.

>        To improve DX in the next iteration we can consider autogeneration of alias with the pattern: `app.your_resource`

Yep, I'd like too, I agree.

Reply:
@lchrusciel Look at the first post where I put some examples